ompi/mca/pml/ob1/pml_ob1_recvfrag.c

```diff
@@ -553,9 +570,44 @@ void mca_pml_ob1_recv_frag_callback_match (mca_btl_base_module_t *btl,
     PERUSE_TRACE_MSG_EVENT(PERUSE_COMM_SEARCH_POSTED_Q_END, comm_ptr,
                            hdr->hdr_src, hdr->hdr_tag, PERUSE_RECV);
 
-    /* release matching lock before processing fragment */
-    OB1_MATCHING_UNLOCK(&comm->matching_lock);
+    /* We matched the frag, Now see if we already have the next sequence in
```

Comment:
Here we are in the match callback of an incoming message, potentially blocking the communication engine. Instead of completing the match for the incoming packet, which would guarantee priority for incoming traffic, the new logic stops after the match but before handling the matched fragment and goes on match and handle an out-of-sequence fragment. This has two issues with me: allows for a period of time to have two fragments matched but not handled and deprioritize the incoming traffic.